src/main/java/mate/academy/spring/controller/OrderController.java

```diff
@@ -0,0 +1,47 @@
+package mate.academy.spring.controller;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import mate.academy.spring.mapper.DtoResponseMapper;
+import mate.academy.spring.model.Order;
+import mate.academy.spring.model.dto.response.OrderResponseDto;
+import mate.academy.spring.service.OrderService;
+import mate.academy.spring.service.ShoppingCartService;
+import mate.academy.spring.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+    private final OrderService orderService;
+    private final UserService userService;
+    private final ShoppingCartService shoppingCartService;
+    private final DtoResponseMapper<OrderResponseDto, Order> orderDtoResponseMapper;
+
+    public OrderController(OrderService orderService,
+                           UserService userService,
+                           ShoppingCartService shoppingCartService,
+                           DtoResponseMapper<OrderResponseDto, Order> orderDtoResponseMapper) {
+        this.orderService = orderService;
+        this.userService = userService;
+        this.shoppingCartService = shoppingCartService;
+        this.orderDtoResponseMapper = orderDtoResponseMapper;
+    }
+
+    @PostMapping("/complete")
+    public OrderResponseDto complete(@RequestParam Long userId) {
+        return orderDtoResponseMapper.toDto(orderService
+                .completeOrder(shoppingCartService.getByUser(userService.get(userId))));
+    }
+
+    @GetMapping
```

Comment:
```suggestion
    @GetMapping("/by-user")
```

Reply:
there was another example in the readme, I can change it if necessary
![image](https://user-images.githubusercontent.com/115244741/235670854-d055a6ab-fd8b-42c0-bcc9-2bbcea377e70.png)